src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
 
+        File sourceFile = new File(fromFileName);
+        StringBuilder builder = new StringBuilder();
+        int supplySum = 0;
+        int buySum = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(sourceFile))) {
+            String value = reader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Source file doesn't exists or not readable", e);
```

Comment:
The error message 'Source file doesn't exists or not readable' is misleading. An IOException can occur for various reasons, not just because the file doesn't exist or isn't readable. Consider using a more general message like 'Error reading from source file'.